component/src/main/java/io/siddhi/extension/io/gcs/sink/internal/strategies/RotateIntervalStrategy.java

```diff
@@ -0,0 +1,54 @@
+package io.siddhi.extension.io.gcs.sink.internal.strategies;
+
+import io.siddhi.extension.io.gcs.sink.internal.beans.GCSSinkConfig;
+import io.siddhi.extension.io.gcs.sink.internal.content.ContentAggregatorFactory;
+import io.siddhi.extension.io.gcs.sink.internal.publisher.PublisherTask;
+import io.siddhi.extension.io.gcs.sink.internal.util.RotationStrategy;
+import java.util.HashMap;
+import java.util.concurrent.Future;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Class to handle span based object rotation
+ */
+public class RotateIntervalStrategy extends RotationStrategy {
+
+    private GCSSinkConfig config;
+    private HashMap<String, Future> scheduledFuturesMap = new HashMap<>();
+
+    public RotateIntervalStrategy(GCSSinkConfig config) {
+        this.config = config;
+    }
+
+    @Override
+    public void queueEvent(String objectName, Object event) {
+
+
+        try {
+            if (getStateContainer().lock()) {
+                if (getStateContainer().getQueuedEventMap().containsKey(objectName)) {
+                    getStateContainer().getQueuedEventMap().get(objectName).addEvent(event);
+                    getStateContainer().getEventOffsetMap().put(objectName,
+                            getStateContainer().getEventOffsetMap().get(objectName).intValue() + 1);
+
+
+                } else {
+                    getStateContainer().getQueuedEventMap().put(objectName,
+                            ContentAggregatorFactory.getContentGenerator(config));
+                    getStateContainer().getQueuedEventMap().get(objectName).addEvent(event);
+                    getStateContainer().getEventOffsetMap().put(objectName, Integer.valueOf(1));
+                }
+            }
+        } finally {
+            getStateContainer().releaseLock();
+        }
+
+        if (!scheduledFuturesMap.containsKey(objectName) || scheduledFuturesMap.get(objectName).isDone()) {
+            scheduledFuturesMap.put(objectName, config.getScheduledExecutorService()
```

Comment:
Shouldn't you stop the scheduled job once flushed by other means

Reply:
For the case of RotateInterval it is not the case as the Sink will keep the file open for a certain time and then it will do the file commit only in CountBased rotation that should be done